libsolidity/parsing/Parser.cpp

```diff
@@ -1050,8 +1051,9 @@ ASTPointer<InlineAssembly> Parser::parseInlineAssembly(ASTPointer<ASTString> con
 	shared_ptr<yul::Block> block = asmParser.parse(m_scanner, true);
 	if (block == nullptr)
 		BOOST_THROW_EXCEPTION(FatalError());
-	nodeFactory.markEndPosition();
-	return nodeFactory.createNode<InlineAssembly>(_docString, block);
+
+	location.end = block->location.end;
```

Comment:
Other code uses `nodeFactory.setEndPositionFromNode(block)`.

Reply:
`block` is `yul::Block`, not `ASTPointer<ASTNode>`

Reply:
Ah, I see!